client/utils/guidance.py

```diff
@@ -136,42 +144,107 @@ def show_guidance_msg(self, unique_id, input_lines, access_token, hash_key, guid
             print(GUIDANCE_DEFAULT_MSG)
             return EMPTY_MISUCOUNT_TGID_PRNTEDMSG
 
-        lst_mis_u = dict_info.get('lstMisU')
+        shorten_unique_id = assess_id_util.canonicalize(unique_id)
+        # Try to get the info dictionary for this question. Maps wrong answer
+        # to dictionary
+        assess_dict_info = self.guidance_json[
+            'dictAssessId2Info'].get(shorten_unique_id)
+        if not assess_dict_info:
+            log.info("shorten_unique_id is not in dictAssessId2Info")
+            print(GUIDANCE_DEFAULT_MSG)
+            return EMPTY_MISUCOUNT_TGID_PRNTEDMSG
+
+        wa_details = assess_dict_info['dictWA2DictInfo'].get(response)
+        if not wa_details:
+            log.info("Cannot find the wrong answer in the WA2Dict for this assesment.")
+            lst_mis_u = None
+        else:
+            lst_mis_u = wa_details.get('lstMisU')
 
         # No list of misunderstandings for this wrong answer, default message
         if not lst_mis_u:
             log.info("Cannot find the list of misunderstandings.")
-            print(GUIDANCE_DEFAULT_MSG)
-            return EMPTY_MISUCOUNT_TGID_PRNTEDMSG
 
-        self.misU_count_dict = self.update_misUcounts(hash_key, lst_mis_u,
-                                                      repr(input_lines), shorten_unique_id)
+        wa_count_threshold = self.guidance_json['wrongAnsThresh']
+        wa_lst_assess_num = assess_dict_info['dictWA2LstAssessNum_WA']
+        msg_id_set = set()
 
-        Wrong_Answer_threshold = self.guidance_json['wrongAnsThresh']
+        answerDict, countData = self.get_misUdata()
+        prev_responses = answerDict.get(shorten_unique_id, [])
 
-        msg_id_set = set()
+        # Confirm that this WA has not been given before
+        seen_before = response in prev_responses
 
-        # If the count is higher than the threshold, we need to display the
-        # message
-        for mis_u in self.misU_count_dict:
```

Comment:
I think one reason why things are confusing is because this loop got removed. This loop should happen after all changes to `self.mis_count_dict` have been done. Which means after the `if not seen_before` conditional.